pages/css-grid-superposition--qcj5jo/css-grid-superposition--qcj5jo.md

```diff
@@ -0,0 +1,279 @@
+---
+title: Prevent layout shift with CSS grid superposition
+date: Last Modified
+layout: article.njk
+---
+
+<!-- TODO still some confusion between "we" and I -->
+
+There are two reasons why people use CSS grid:
+
+1. 😎 CSS is awesome! It's a fact, deal with it. 
+1. 🛠️ Grid is a great tool to build complex two-dimensional layouts.
+
+I sometimes have a third reason to use CSS grid, I call it the "anti layout shift superposition technique".
+Let me explain with some real examples.
+
+**NOTE:** This article is focused on layout shift occuring when toggling state on components.
+We won't cover layout shift happening on first page load.
+
+## Let's get some context
+
+I work for [Clever Cloud](https://www.clever-cloud.com/en/), an IT automation platform.
+Our clients push their code and we handle the rest for them: build, deploy, host, scale, maintain, recover...
+When they need to configure their applications, databases and other services, they use our web UI: "the console".
+I spend most of my time on this project and it looks like this:
+
+![Console overview of a Node.js application](./console-overview.png)
+
+What you see in the screenshot above is the overview of my own site, a Node.js application hosted on our platform.
+On the right, you can see a pie chart which represents the repartition of HTTP response codes returned by the application in the last 24 hours.
+
+![UI Component displaying a chart](./cc-tile-status-codes-chart.png)
+
+<!-- TODO -->
+When I worked on this component, I imagined our users would implicitly understand the chart is related to the current application.
+I decided to move some details like the exact number of requests in tooltips accessible via mouse and touch.
+The problem was, how do I explain the following details:
+
+* The chart only displays data from the last 24 hours.
+* The legend items can be clicked to show/hide the different status code categories. Thank you [Chart.js](https://www.chartjs.org/)!
+
+The available space was too small to add a title at the top or bottom of the chart so to help our users figure this out, I added an information button ℹ️ in the top right corner.
+When they click on it, a short text is displayed in place of the chart like this:
+
+![UI Component displaying some help](./cc-tile-status-codes-help.png)
+
+When they click on the cross icon, we hide the short text and display the chart again.
+Implementing this toggle behaviour was not that hard.
```

Comment:
"was" -> "is" ? That way you include the reader in your journey. (Or so I see it 😅)

Reply:
Agreed. To me "was" means that you're just stating a fact and will not talk about this further. With "is", I expect that you are going to talk about it and explain how to implement this. And... you are so "is" works well here.

Reply:
```suggestion
Implementing this toggle behaviour is not that hard.
```